exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/PluginConfigWrapper.java

```diff
@@ -114,4 +125,75 @@ public boolean isOauth() {
 
     return tokenCredentials.map(OAuthTokenCredentials::getClientID).orElse(null) != null;
   }
+
+  @JsonIgnore
+  public String getClientID() {
+    CredentialedStoragePluginConfig securedStoragePluginConfig = (CredentialedStoragePluginConfig) config;
+    CredentialsProvider credentialsProvider = securedStoragePluginConfig.getCredentialsProvider();
+
+    return credentialsProvider.getCredentials().getOrDefault("clientID", "");
+  }
+
+  /**
+   * This function generates the authorization URI for use when a non-admin user is authorizing
+   * OAuth2.0 access for a storage plugin.  This function is necessary as we do not wish to expose
+   * any plugin configuration information to the user.
+   *
+   * If the plugin is not OAuth, or is missing components, the function will return an empty string.
+   * @return The authorization URI for an OAuth enabled plugin.
+   */
+  @JsonIgnore
+  public String getAuthorizationURIWithParams() {
```

Comment:
Just a note that I'll try to move this code, probably into OAuthTokenCredentials or the storage config, in DRILL-8200.

Reply:
Sounds good.  Just be aware that this code is used in the `list.ftl` for the Credentials page.